packages/aws-cdk/lib/api/aws-auth/awscli-compatible.ts

```diff
@@ -149,7 +149,12 @@ export class AwsCliCompatible {
  */
 async function getRegionFromIni(profile: string): Promise<string | undefined> {
   const sharedFiles = await loadSharedConfigFiles({ ignoreCache: true });
-  return sharedFiles?.configFile?.[profile]?.region || sharedFiles?.configFile?.default?.region;
+  // https://docs.aws.amazon.com/cli/v1/userguide/cli-configure-files.html
+  return getRegionFromIniFile(profile, sharedFiles.configFile) ?? getRegionFromIniFile(profile, sharedFiles.credentialsFile);
```

Comment:
I'm not sure what takes precedence: `config` or `credentials`? Looking it up but if anybody knows do tell.

Note that in `config` file, profiles are suffixed with the `profile ` string. Nonetheless, you the SDK parses it, it ends having the same keys as the `credentials` file, so there's no need to add the prefix when querying the file.

```console
ConfigFile: {
  "default": {
    "region": "us-east-2"
  },
  "user1": {
    "region": "us-east-2"
  }
}
CredentialsFile: {
  "default": {
    "region": "us-east-2"
  },
  "user1": {
    "region": "us-east-2"
  }
}
```

For files that look like: 

```console
❯ cat ~/.aws/config                                                                                                                                                                                                                             [13:52:11]
[default]
region=us-east-2

[profile user1]
region=us-east-2

 ❯ cat ~/.aws/credentials                                                                                                                                                                                                                        [13:52:16]
[default]
region=us-east-2

[user1]
region=us-east-2
```

Reply:
I was thinking of the order:

- Profile config file
- Profile credentials file
- Default config file
- Default credentials file

So profile strictly before default.

Thoughts?